NuRadioReco/detector/RNO_G/rnog_detector.py

```diff
@@ -0,0 +1,1488 @@
+
+        _gain = np.abs(self(freqs))
+        if in_dB:
+            mask = _gain > 0  # to avoid RunTime warning
+            ax.plot(freqs[mask] / units.MHz, 20 * np.log10(_gain[mask]), color="k", label="total")
+        else:
+            ax.plot(freqs / units.MHz, _gain, color="k", label="total")
+
+        ax.set_xlabel("frequency / MHz")
+        if in_dB:
+            ax.set_ylabel("gain / dB")
+        else:
+            ax.set_ylabel("gain")
+            ax.set_yscale("log")
+
+        ax.legend()
+
+        if show:
+            plt.show()
+        else:
+            return fig, ax
+
+    def get_time_delay(self):
+        """ Get time delay from DB """
+        return np.sum(self.__time_delays)
+
+    def _get_time_delay(self):
+        """
+        Calculate time delay from phase of the stored complex response function.
+        This is not the time delay which is stored in the DB and which is used in
+        the `__init__()` to normalize the reponse function. Rather, its the remaining
+        group delay.
+
+        The time delay is calculated as the mean between 195 and 205 MHz.
+
+        Returns
+        -------
+
+        time_delay1 : float
+            The time delay at ~ 200 MHz
+        """
+
+        freqs = np.arange(0.05, 1.2, 0.001) * units.GHz
+
+        response = self(freqs)
+
+        delta_freq = np.diff(freqs)
+
+        phase = np.angle(response)
+        time_delay = -np.diff(np.unwrap(phase)) / delta_freq / 2 / np.pi
+
+        mask = np.all([195 * units.MHz < freqs, freqs < 250 * units.MHz], axis=0)[:-1]
+        time_delay1 = np.mean(time_delay[mask])
+
+        # fit the unwrapped phase with a linear function
+        popt = np.polyfit(freqs, np.unwrap(phase), 1)
+        time_delay2 = -popt[0] / (2 * np.pi)
+
+        if np.abs(time_delay1 - time_delay2) > 0.1:
+            logging.warning("Calculation of time delay. The two methods yield different results: "
+                            f"{time_delay1:.1f} / {time_delay2:.1f} for {self.get_names()}. Return the former...")
+
+        return time_delay1
+
+
+
+if __name__ == "__main__":
+
+    from NuRadioReco.detector import detector
+
+    det = detector.Detector(source="mongo", log_level=logging.DEBUG, always_query_entire_description=False,
+                            database_connection='RNOG_public', select_stations=24)
+
+    det.update(datetime.datetime(2023, 8, 2, 0, 0))
+    # det.export("station24.json.xz")
+    # # print(det.get_absolute_position(21))
+    # # resp = det.get_signal_chain_response(21, 0)
+    # # resp.plot(True, True)
+
+    # det = detector.Detector(source="mongo", detector_file="station24.json.xz", select_stations=24)
+    # det.update(datetime.datetime(2023, 8, 2, 0, 0))
+
+    response = det.get_signal_chain_response(24, 0)
+    print(response)
+
+    # from NuRadioReco.modules.io import eventReader
+
+    # reader = eventReader.eventReader()
+    # reader.begin("./NuRadioMC/test/SingleEvents/1e18_output.nur")
```

Comment:
I think it's fine to have a test script in here, but if you want to keep it maybe tidy up the commented part before merging?

Reply:
sure that is for R&D and gets removed before merging

Reply:
done